docs/migration/4_0.md

```diff
@@ -1,3 +1,25 @@
 # Technical Changes in Spartacus 4.0
 
 ## Breaking Changes Introduced in 4.0
+
+### Page resolvers
+In 3.1 and 3.2 we've introduced a few changes on how the page meta data is collected. The resolvers are now configurable and whether they render on the client or server (SSR) is also configurable. A few resolvers are changed or added, since most data is now configurable in the backend (description, robots). The robot information that was previously hardcoded in some resolvers, are now driven by backend data. 
+
+A new feature has landed in the product and category resolvers for the canonical URL.
+
+The `BasePageMetaResolver` is leveraged to compose most of the generic page meta data. Most page resolvers now use the page data to create the description and robot tags. The changes have affected the following resolver classes:
+- The `PageMetaService` has a dependency on `UnifiedInjector`, `PageMetaConfig` and the `platformId`.
+- The `ContentPageMetaResolver` depends only on the `BasePageMetaResolver`.
+- The `BasePageMetaResolver` requires the `Router` and `PageLinkService` to add canonical links to the page meta.
+- The `ProductPageMetaResolver` requires the `BasePageMetaResolver` and `PageLinkService`.
+- The `CategoryPageMetaResolver` requires the `BasePageMetaResolver`.
+- The `SearchPageMetaResolver` requires the `BasePageMetaResolver`.
+- The `CartPageMetaResolver` depends only on the  `BasePageMetaResolver`.
```

Comment:
It should be mentioned here that we removed `CartPageMetaResolver`, because it worked like `ContentPageMetaResolver`